### PR TITLE
gromacs: add fast math flags when AOCC

### PR DESCRIPTION
**Pull request by amd-toolchain-support**
Adding Optimization flags `-Ofast` `-ffast-math` for AOCC compiler

### CODE REVIEW

File: var/spack/repos/builtin/packages/gromacs/package.py [hunk 1]
```diff
@@ -256,6 +256,11 @@ def cmake_args(self):
                 '-DGMX_MPI:BOOL=OFF',
                 '-DGMX_THREAD_MPI:BOOL=ON'])
 
+        if self.spec.satisfies('%aocc'):
+            opt_flags = '-Ofast -ffast-math'
+            options.append(self.define('CMAKE_C_FLAGS_RELEASE', opt_flags))
```
Comment (marvinbernhardt) [inline var/spack/repos/builtin/packages/gromacs/package.py:261]: Note that the default build type is RelWithDebInfo, so probably that should get the those flags, too.

**Review by marvinbernhardt — CHANGES_REQUESTED** [review]
Otherwise looks good :+1:

File: var/spack/repos/builtin/packages/gromacs/package.py (hunk 1 above)
Comment (amd-toolchain-support) [reply var/spack/repos/builtin/packages/gromacs/package.py:261]: Hi @marvinbernhardt , RelWithDebInfo picks `-O2` by default for debugging purpose, we don't want to pass `-Ofast` for Debug build.

Comment (marvinbernhardt) [reply var/spack/repos/builtin/packages/gromacs/package.py:261]: Ah ok, did'nt know that!

**Review by marvinbernhardt — DISMISSED** [review]
LGTM

**Comment by amd-toolchain-support** [comment]
Hi @adamjstewart ,

Can you please review and do the needful for this request.

Thanks.

File: var/spack/repos/builtin/packages/gromacs/package.py (hunk 1 above)
Comment (haampie) [reply var/spack/repos/builtin/packages/gromacs/package.py:261]: Do you really need -ffast-math? It's known to lead to all sorts of problems. https://simonbyrne.github.io/notes/fastmath/.

Something tells me this is just for benchmark reports somewhere... If that's the use case, then note you can inject those flags like `spack install gromacs cflags=-ffast-math` and you don't have to force them onto users. It's bad practice IMHO.

**Review by becker33 — CHANGES_REQUESTED** [review]
This isn't the sort of change that should be compiler-dependent. Either `gromacs` is stable under `-ffast-math` and it should be enabled for every compiler, or it's unsafe and we should not be enabling it by default.

As @haampie wrote in another channel:
> the main issue with -ffast-math is that it can influence computations in dependent libraries and executables because it enables flush to zero as a cpu mode. So if you just dlopen(...) a library compiled with ffast-math, the computations in your executable now flush subnormals.

File: var/spack/repos/builtin/packages/gromacs/package.py (hunk 1 above)
Comment (junghans) [reply var/spack/repos/builtin/packages/gromacs/package.py:261]: I agree, just use `cflags=...` for special OMG fast builds.

**Comment by amd-toolchain-support** [comment]
Hi @becker33 , we have removed `-ffast-math` flag from this pull request, please do the needful.

**Comment by haampie** [comment]
`-Ofast` is a superset of `-ffast-math`...

**Comment by amd-toolchain-support** [comment]
@haampie , thanks for the comment, we have understood the concern from the community hence considering to drop the idea of using `-ffast-math` in our recipe.

Thanks.

**Pull request closed without merging**